### PR TITLE
[WIP] Changing the way headers are handled

### PR DESCRIPTION
**Pull request by tbrown1979**
This PR eliminates the need for headers to be passed everywhere. The `headersToSearchFor` value was removed from the CSVParser, and then I collapsed the parsing functionality into the object. 

Before this PR - we were taking in headers as a parameter and verifying that they existed in the given csv file. 
Now - we just grab all the headers that appear in the csv file (without passing in a parameter) and all the corresponding fields. Only when you go to convert will it fail because a field may be missing. I collapsed the parsing functionality into the CSVParser object. Keeping it as a class seemed unnecessary.

Also, I added in a new CSVParsingError called `EmptyHeadersFound` that is thrown when a CSV file's headers contains an empty string.

### CODE REVIEW

File: src/main/scala/CSVParser.scala [hunk 1]
```diff
@@ -42,22 +36,21 @@ class CSVParser(headersToSearchFor: Seq[String]) {
...
-      lazy val headers = maybeHeaders.getOrElse(Seq.empty[String]).map(normalizeString(_))
-      val fieldIndexes: Seq[Tuple2[String, Int]] = findFieldIndexes(headers)
+      val headers = maybeHeaders.getOrElse(Seq.empty[String]).map(normalizeString(_))
 
-      if (fieldIndexes.size < headersToSearchFor.size) {
-        val missingHeaders = headersToSearchFor.diff(fieldIndexes.map(_._1)).map(_.toString)
-        Failure(NonEmptyList(MissingHeaders(missingHeaders)))
-      } else {
+      if (headers.exists(_ == ""))
+        fail(EmptyHeadersFound(headers))
```
Comment (adamdecaf) [inline src/main/scala/CSVParser.scala]: Should we do `_.trim == ""`?

Comment (tbrown1979) [inline src/main/scala/CSVParser.scala]: They get normalized(trimmed) above so it isn't necessary here. Just added a test to make sure that it was working correctly.

Comment (adamdecaf) [inline src/main/scala/CSVParser.scala]: Ok, how about calling `.isEmpty` instead?

Comment (tbrown1979) [inline src/main/scala/CSVParser.scala]: Definitely better! Got that changed

**Comment by tbrown1979** [comment]
retest this please

**Comment by tbrown1979** [comment]
Anybody know why this is failing?

**Comment by zmccoy** [comment]
sha1 issue, I'm not sure what causes this.  @adamdecaf @DustinChaloupka Do either of you know what causes this issue to happen?

**Comment by zmccoy** [comment]
retest this please

**Comment by adamdecaf** [comment]
@zmccoy No, the last time I looked there was a bug in the jenkins plugin that wouldn't handle this properly.

**Comment by adamdecaf** [comment]
:+1:

**Comment by zmccoy** [comment]
:+1:

**Pull request merged**